server/plugin/command.go

```diff
@@ -501,6 +518,163 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
 	}
 }
 
+func (p *Plugin) handlewebhook(_ *plugin.Context, args *model.CommandArgs, parameters []string, userInfo *GitHubUserInfo) string {
+	if len(parameters) == 0 {
+		return "Invalid webhook command. Available command are `add` and `list`."
+	}
+	command := parameters[0]
+	parameters = parameters[1:]
+	ctx := context.Background()
+	githubClient := p.githubConnectUser(ctx, userInfo)
+	baseURL := p.getBaseURL()
+	switch command {
+	case add:
+		if len(parameters) < 1 {
+			return "Invalid parameter for add command, provide repo details in `[owner/repo]` format."
+		}
+
+		owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+		siteURL := *p.API.GetConfig().ServiceSettings.SiteURL + Manifest.Props["inbound_webhook_url"].(string)
+		var hook github.Hook
+		var config = make(map[string]interface{})
+		config["secret"] = p.getConfiguration().WebhookSecret
+		if len(parameters) >= 1 {
+			parameters = parameters[1:]
+			for _, val := range parameters {
+				switch {
+				case strings.Contains(val, "https://"):
+					config["url"] = val
+				case strings.Contains(val, "true") || strings.Contains(val, "false"):
+					insecureVal, _ := strconv.ParseBool(val)
+					config["insecure_ssl"] = insecureVal
+				case strings.Contains(val, "application/json") || strings.Contains(val, "x-www-form-urlencoded"):
+					config["content_type"] = val
+				default:
+					if strings.Contains(val, "*") && len(val) == 1 {
+						hook.Events = []string{"*"}
+						break
+					}
+					if err := p.CheckOptionsValid(strings.Split(val, ",")); err != nil {
+						return err.Error()
+					}
+					hook.Events = strings.Split(val, ",")
+				}
+			}
```

Comment:
@mickmister tried with slash command but it was too confusing for user i felt

currently we need 5 parameters from user
[org/repo]  - 
callBackUrl - optional ( default to MMURL +"plugin/github/plugin")
content-type - optional ( default x-www-form-urlencoded)
event - optional ( * -default , create ,delete .... ) 
insecureSSL -  optional ( true / false , default to false) - already implemented with flags

but while parsing instead of taking in sequence as there are lot of parameter so tried this way,
if needed can do otherway , please let me know your view

Reply:
> tried with slash command but it was too confusing for user i felt

Confusing in what way?

> currently we need 5 parameters from user

Since 4 are optional, I would say we only need 1, but that is beside the point. 2/5 we should use flags instead of guessing what field they are filling out. The required one can be positional if that is possible, but all the others should be flag arguments.